include/mlboard_logger.hpp

```diff
@@ -0,0 +1,200 @@
+   * @param epochCount Interval of epochs youw want to log your data.
+   * @param summaryTag Tag to use for summary.
+   * @param summaryType Type of summary to be logged.
+   * @param embeddingMetadata Metadata for embedding.
+   * @param heighofImage The height of image to be logged.
+   * @param widthofImage The width of image to be logged.
+   */
+  MlboardLogger(
+      mlboard::FileWriter& output,
+      std::function<arma::mat()> func,
+      std::string summaryTag,
+      std::string summaryType,
+      std::vector<std::string> embeddingMetadata = {},
+      size_t heightofImage = 0,
+      size_t widthofImage = 0,
+      int epochCount = 1)
+    : callbackUsed(true),
+      output(output),
+      matFunc(func),
+      widthofImage(widthofImage),
+      heightofImage(heightofImage),
+      embeddingMetadata(embeddingMetadata),
+      summaryType(summaryType),
+      epochCount(epochCount),
+      summaryTag(summaryTag)
+  {
+    // Nothing to do here.
+  }
+
+  /**
+   * Callback function called at the end of a pass over the data.
+   *
+   * @param optimizer The optimizer used to update the function.
+   * @param function Function to optimize.
+   * @param coordinates Starting point.
+   * @param epoch The index of the current epoch.
+   * @param objective Objective value of the current point.
+   */
+  template<typename OptimizerType, typename FunctionType, typename MatType>
+  void EndEpoch(OptimizerType& /* optimizer */,
+                FunctionType& /* function */,
+                const MatType& /* coordinates */,
+                const size_t epoch,
+                double objective)
+  {
+    if (callbackUsed && summaryType == "")
+    {
+      objective = localFunc();
+    }
+    if (summaryType == "")
+    {
+      if (epoch % epochCount == 0)
+      {
+        mlboard::SummaryWriter<mlboard::FileWriter>::Scalar(lossTag,
+            epoch / epochCount, objective, output);
+        mlboard::SummaryWriter<mlboard::FileWriter>::Scalar(accTag,
+            epoch / epochCount, 1 - objective, output);
+      }
+    }
+    if (callbackUsed && (epoch % epochCount == 0) && summaryType != "")
+    {
+      arma::mat valueToBeLogged = matFunc();
+      if (summaryType == "histogram")
+      {
+        mlboard::SummaryWriter<mlboard::FileWriter>::Histogram(summaryTag,
+            epoch / epochCount, arma::rowvec(valueToBeLogged), output);
+      }
+      else if (summaryType == "embedding")
+      {
+        mlboard::SummaryWriter<mlboard::FileWriter>::Embedding(summaryTag,
+            valueToBeLogged, embeddingMetadata, output);
+      }
+      else if (summaryType == "image")
+      {
+        // channel is not needed to log image
+        mlpack::data::ImageInfo info(widthofImage, heightofImage, 0);
+        mlboard::SummaryWriter<mlboard::FileWriter>::Image(
+              summaryTag, epoch / epochCount , valueToBeLogged,
+              info, output);
+      }
+      else
+      {
+        /* Do nothing */
```

Comment:
I think instead of leaving it like this, we can raise an error that this summary type is not supported by the callback.